backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -805,6 +821,96 @@ private CommitEntity getCommitEntityObj(
     }
   }
 
+  @Override
+  public VersioningLineageEntry getVersioningEntryByBlob(
+      Session session,
+      RepositoryFunction repositoryFunction,
+      VersioningBlobElement versioningBlobElement)
+      throws ModelDBException {
+    RepositoryEntity repositoryEntity = repositoryFunction.apply(session);
+    String blobSha = versioningBlobElement.getBlobSha();
+    String blobType = versioningBlobElement.getBlobType();
+    List list = getInternalFolderElementEntity(session, blobSha, blobType);
+    for (Object entity : list) {
+      InternalFolderElementEntity internalFolderElementEntity =
+          (InternalFolderElementEntity) entity;
+      VersioningLineageEntry result =
+          getCommit(
+              session,
+              internalFolderElementEntity,
+              VersioningLineageEntry.newBuilder()
+                  .addLocation(internalFolderElementEntity.getElement_name()),
+              repositoryEntity);
+      if (result != null) {
+        return result;
+      }
```

Comment:
What is this doing? Looks weird to me.

Reply:
That runs the code that starts from a specified blob hash and type entry and finds required commit hash + location